src/hooks/useTaskStorage.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react';
+import { TaskItemType } from '../types/TaskItemType';
+
+
+export const useTaskStorage = () => {
+  const [taskList, setTaskList] = useState<TaskItemType[]>(
+    JSON.parse(localStorage.getItem('tasks-list')) || [],
+  );
+
+  useEffect(() => {
+    localStorage.setItem('tasks-list', JSON.stringify(taskList));
```

Comment:
We should not set up localStorage inside useEffect,, here suppose to be only getItems - but you already have it inside useState... 
What will happen from line 5 top line 12 inside this hook actually ?